### PR TITLE
Inventory Change

### PR DESCRIPTION
**Pull request by yingwei123**
Added GUI for inventory -> Click "i" to toggle
Created Inventory Class and replaced old inventory in player.gd

### CODE REVIEW

File: scenes/CanvasLayer.gd [hunk 1]
```diff
@@ -0,0 +1,20 @@
+extends CanvasLayer
```
Comment (jma8774) [inline scenes/CanvasLayer.gd]: Surely there's a better file name than CanvasLayer.gd

File: scenes/CanvasLayer.gd [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+
+@onready var inventory = $InventoryGui
+
+# Called when the node enters the scene tree for the first time.
+func _ready():
+	inventory.close() # Replace with function body.
```
Comment (BrianCheung1) [inline scenes/CanvasLayer.gd]: can toggle inventory to be auto hidden on level 1 scene so don't need this inventory.close

File: scenes/CanvasLayer.gd [hunk 3]
```diff
@@ -0,0 +1,20 @@
...
+
+# Called every frame. 'delta' is the elapsed time since the previous frame.
+func _process(delta):
+	pass
+
+func _input(event):
+	if event.is_action_pressed("toggle_inventory"):
+		if inventory.isOpen:
```
Comment (BrianCheung1) [inline scenes/CanvasLayer.gd]: instead of all these variables you can also just hide the canvas layer
canvas_layer.visible = !canvas_layer.visible

**Review by BrianCheung1 — COMMENTED** [review]
Check comments

File: scenes/CanvasLayer.gd (hunk 2 above)
Comment (yingwei123) [reply scenes/CanvasLayer.gd]: ima keep this in for now i might change it in a later branch

**Pull request merged**